### PR TITLE
Bugfix/borrow exceeded limit fix

### PR DESCRIPTION
**Pull request by mn13**

### CODE REVIEW

File: pool/src/lib.rs [hunk 1]
```diff
@@ -941,6 +945,113 @@ impl LendingPoolTrait for LendingPool {
...
+        let amount_of_debt_token = debt_coeff
+            .recip_mul_int(amount)
+            .ok_or(Error::MathOverflowError)?;
+        let debt_token_supply_after = debt_token_supply
+            .checked_add(amount_of_debt_token)
+            .ok_or(Error::MathOverflowError)?;
+
+        let amount_to_sub = amount.checked_neg().ok_or(Error::MathOverflowError)?;
+
+        add_token_balance(
```
Comment (artur-abliazimov) [inline pool/src/lib.rs:1013]: There also must be a call to 
```
add_token_total_supply(&env, &reserve.debt_token_address, amount_of_debt_token).
```

File: pool/src/lib.rs [hunk 2]
```diff
@@ -941,6 +945,113 @@ impl LendingPoolTrait for LendingPool {
...
+            &env,
+            &reserve.debt_token_address,
+            &who,
+            amount_of_debt_token,
+        )?;
+        add_token_balance(&env, &asset, &who, amount)?;
+        add_token_balance(&env, &asset, &env.current_contract_address(), amount_to_sub)?;
```
Comment (artur-abliazimov) [inline pool/src/lib.rs:1020]: Could you explain what it is for?

File: pool/src/lib.rs [hunk 3]
```diff
@@ -941,6 +945,113 @@ impl LendingPoolTrait for LendingPool {
...
+            MintBurn::new(
+                AssetBalance::new(reserve.debt_token_address, amount_of_debt_token),
+                true,
+                who.clone(),
+            ),
+            MintBurn::new(AssetBalance::new(asset.clone(), amount), true, who),
+            MintBurn::new(
+                AssetBalance::new(asset, amount),
+                false,
+                env.current_contract_address(),
```
Comment (artur-abliazimov) [inline pool/src/lib.rs:1050]: Should we replace it with `reserve.s_token_address`?

File: pool/src/lib.rs (hunk 1 above)
Comment (mn13) [reply pool/src/lib.rs:1013]: It's done in 1037

**Pull request merged**